Format with black

martymichal opened the pull request:
Helps with #97.

JiriPavela commented:
Good job, please fix the typing errors first and then I will look into the individual files and see whether we need to custom-format (`# fmt: off`) some code constructs (typically collections of large number of constants, etc.).

martymichal committed:
Format whole codebase with Black

Also, add a note about the tool in the CONTRIBUTING guidelines.

JiriPavela committed:
Add black configuration to pyproject.toml

The line length parameter has been set to 100, as this is the line
length we have been using throughout the development of Perun.

JiriPavela committed:
Fix typing error caused by moved type: ignore

JiriPavela committed:
Handle too long strings using experimental flag

Base black configuration does not format strings that are too long.
A `--experimental-string-processing` flag has been used to format
them properly.

JiriPavela committed:
Resolve Codacy issues after black formatting

JiriPavela reviewed (approved):
I polished the PR and I think we are ready for a merge. There might still be some code blocks that were perhaps formatted too aggressively, but we can solve those ad-hoc as we spot them.

Also, there are some Codacy issues that I believe were in the codebase before this PR, but they were likely triggered again by the formatting changes. I would leave fixing those issues for the #133.

tfiedor commented on docs/_static/templates/supported_module_names.py:
```diff
@@ -1,12 +1,20 @@
+        "vcs": ["git"],
+        "collect": ["trace", "memory", "time"],
+        "postprocess": ["filter", "normalizer", "regression-analysis"],
+        "view": [
```
I don't really like how black formatted this.
